src/lib/snack-bar/snack-bar-container.ts

```diff
@@ -11,6 +18,20 @@ import {
 } from '../core';
 import {MdSnackBarConfig} from './snack-bar-config';
 import {MdSnackBarContentAlreadyAttached} from './snack-bar-errors';
+import {Observable} from 'rxjs/Observable';
+import {Subject} from 'rxjs/Subject';
+
+
+export class SnackBarState {
```

Comment:
Can we do this with a string literal type? E.g.,
```ts
type SnackBarState = 'initial' | 'visibile' | 'complete' | 'void';
``` 

I think this would make the animation definitions much cleaner.